test/algorithms/classifiers/test_vqc.py

```diff
@@ -14,184 +14,141 @@
-        y = 1.0 * (np.sum(X, axis=1) <= 1)
-        while len(np.unique(y)) == 1:
-            X = algorithm_globals.random.random((num_samples, num_inputs))
-            y = 1.0 * (np.sum(X, axis=1) <= 1)
-        y = np.array([y, 1 - y]).transpose()  # VQC requires one-hot encoded input
+        x = np.asarray(self.binary_data["X"])
+        y = np.asarray(self.binary_data["y"])
 
-        # fit to data
-        classifier.fit(X, y)
+        classifier.fit(x, y)
 
-        # score
-        score = classifier.score(X, y)
-        self.assertGreater(score, 0.5)
+        score = classifier.score(x, y)
+        self.assertGreaterEqual(score, 0.5)
 
     @data(
-        # num_qubits, feature_map, ansatz
-        (True, False, False),
-        (True, True, False),
-        (True, True, True),
-        (False, True, True),
-        (False, False, True),
-        (True, False, True),
-        (False, True, False),
+        # optimizer, quantum instance
+        (cobyla, sv_quantum_instance),
+        (cobyla, qasm_quantum_instance),
+        (bfgs, sv_quantum_instance),
+        (bfgs, qasm_quantum_instance),
+        (None, sv_quantum_instance),
+        (None, qasm_quantum_instance),
     )
-    def test_default_parameters(self, config):
-        """Test VQC instantiation with default parameters."""
+    def test_vqc(self, config):
+        """Test VQC."""
+        optimizer, quantum_instance = config
 
-        provide_num_qubits, provide_feature_map, provide_ansatz = config
         num_inputs = 2
+        feature_map = ZZFeatureMap(num_inputs)
+        ansatz = RealAmplitudes(num_inputs, reps=1)
 
-        num_qubits, feature_map, ansatz = None, None, None
-
-        if provide_num_qubits:
-            num_qubits = num_inputs
-        if provide_feature_map:
-            feature_map = ZZFeatureMap(num_inputs)
-        if provide_ansatz:
-            ansatz = RealAmplitudes(num_inputs, reps=1)
-
-        classifier = VQC(
-            num_qubits=num_qubits,
-            feature_map=feature_map,
-            ansatz=ansatz,
-            quantum_instance=self.qasm_quantum_instance,
-        )
-
-        # construct data
-        num_samples = 5
-        # pylint: disable=invalid-name
-        X = algorithm_globals.random.random((num_samples, num_inputs))
-        y = 1.0 * (np.sum(X, axis=1) <= 1)
-        while len(np.unique(y)) == 1:
-            X = algorithm_globals.random.random((num_samples, num_inputs))
-            y = 1.0 * (np.sum(X, axis=1) <= 1)
-        y = np.array([y, 1 - y]).transpose()  # VQC requires one-hot encoded input
-
-        # fit to data
-        classifier.fit(X, y)
+        initial_point = np.array([0.5] * ansatz.num_parameters)
 
-        # score
-        score = classifier.score(X, y)
-        self.assertGreater(score, 0.5)
+        x = np.asarray(self.binary_data["X"])
+
+        for y in ["y", "y_one_hot", "y_str"]:
+            with self.subTest(f"{y}"):
```

Comment:
Considering that you have @<!-- -->data, I'd suggest to move the `for` loop to @<!-- -->data and to avoid long declarations you can use `itertools.product` like here:
```python
    @idata(itertools.product(OPTIMIZERS, L1L2_ERRORS, QUANTUM_INSTANCES, CALLBACKS))
    @unpack
    def test_classifier_with_opflow_qnn(self, opt, loss, q_i, cb_flag):
```

Reply:
That's a nice suggestion. Thanks! Will do.